spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/ConfigServerConfigDataLocationResolver.java

```diff
@@ -126,20 +125,16 @@ protected PropertyHolder loadProperties(ConfigDataLocationResolverContext contex
 			// from the properties from the context. These are set in
 			// ConfigServerInstanceMonitor.refresh which will only
 			// be called the first time we fetch configuration.
+			// So here we should get the ConfigClientProperties from ConfigServerInstanceMonitor
 			if (discoveryEnabled) {
-				ConfigClientProperties bootstrapConfigClientProperties = context.getBootstrapContext()
-					.get(ConfigClientProperties.class);
-
+				ConfigServerInstanceMonitor configServerInstanceMonitor = context.getBootstrapContext()
+						.get(ConfigServerInstanceMonitor.class);
+				ConfigClientProperties bootstrapConfigClientProperties = configServerInstanceMonitor.getConfig();
```

Comment:
I assume the ConfigClientProperties here is the same as the one in the BootstrapContext?  The comment above indicates it should come from there.

Reply:
That's different, because once ConfigClientProperties first fetch from BootstrapContext then set in the new ConfigClientProperties#uri then the new ConfigClientProperties will replace the old ConfigClientProperties in BootstrapContext. Next time when you invoke RefreshEndpoint for refresh envirment. ConfigServerConfigDataLocationResolver#resolveProfileSpecific, you will fetch the ConfigClientProperties from BootstrapContext but this bean is not the ConfigClientProperties in ConfigServerInstanceMonitor so the ConfigClientProperties in BootstrapContext will never be affected by ConfigServerInstanceMonitor so ConfigClientProperties.uri is wrong when eureka find some instance of cloud-config-server is down and refresh the ConfigClientProperties.uri

Reply:
I understand all of that, but I want to make sure the behavior before the uri is updated is maintained.

I want to confirm that `configServerInstanceMonitor.getConfig()` returns the same instance as `context.getBootstrapContext().get(ConfigClientProperties.class)` currently does.

Reply:
![image](https://github.com/user-attachments/assets/89b54b35-dfe0-431d-8322-9c196fbdfd10)
![image](https://github.com/user-attachments/assets/5e88cb66-d768-473a-8ed5-69673545b179)

Reply:
> I understand all of that, but I want to make sure the behavior before the uri is updated is maintained.
> 
> I want to confirm that `configServerInstanceMonitor.getConfig()` returns the same instance as `context.getBootstrapContext().get(ConfigClientProperties.class)` currently does.

I think you need make some change in this line, the detail code I need read more code about Binder.
![image](https://github.com/user-attachments/assets/15658f41-7804-4276-b747-abfaec686d2e)

Reply:
Please use github to add comment and do not send images it makes it hard to keep track of everything....

So you are saying this PR likely needs more changes then?

Reply:
because I see you want to keep the ConfigClientProperties in configServerInstanceMonitor and BootstrapContext is the same object, so I think ConfigServerConfigDataLocationResolver#resolveProfileSpecific shoud instead of create a new ConfigClientProperties but get ConfigClientProperties from BootstrapContext if it is exist.
I need more time to implement that.